### PR TITLE
Update logging with latest generator and GAX.

### PR DESCRIPTION
**Pull request by shinfan**
@garrettjonesgoogle @michaelbausor

### CODE REVIEW

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 1]
```diff
@@ -222,10 +295,9 @@ public void close() throws IOException {
    *
```
Comment (michaelbausor) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Do we want to be generating method samples?

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 2]
```diff
@@ -48,33 +49,88 @@
...
 /**
  * Service Description:
  *
+ * <p>This class provides the ability to make remote calls to the backing service through method
+ * calls that map to API methods. Sample code to get started:
+ *
+ * <pre>
+ * <code>
+ * try (ConfigServiceV2Api configServiceV2Api = ConfigServiceV2Api.createWithDefaults()) {
+ *   String projectName = "";
```
Comment (michaelbausor) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Should this be a formatted name?

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: I think we should.

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 3]
```diff
@@ -48,33 +49,88 @@
...
+ *
+ * <pre>
+ * <code>
+ * ConfigServiceV2Settings configServiceV2Settings = ConfigServiceV2Settings.defaultBuilder()
+ *     .provideChannelWith(myCredentials)
+ *     .build();
+ * ConfigServiceV2Api configServiceV2Api = ConfigServiceV2Api.create(configServiceV2Settings);
+ * </code>
+ * </pre>
+ *
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Why is the naming of classes in `logging` different from the naming of classes in `pubsub`?

In `pubsub` we have 
- `PublisherApi`, `SubscriberApi`
- `PublisherSettings`, `SubscriberSettings`

In `logging`:
- `ConfigServiceV2Api`, `LoggingServiceV2Api`, `MetricsServiceV2Api`
- `ConfigServiceV2Settings`, `LoggingServiceV2Settings`, `MetricsServiceV2Settings`

Is the `V2` in the name really necessary?

File: gcloud-java-logging/pom.xml [hunk 4]
```diff
@@ -1,31 +1,22 @@
...
     Java idiomatic client for Google Cloud Logging.
   </description>
-  <parent>
-    <groupId>com.google.gcloud</groupId>
-    <artifactId>gcloud-java-pom</artifactId>
-    <version>0.1.4</version>
-  </parent>
+  <groupId>com.google.gcloud</groupId>
+  <artifactId>gcloud-java-logging</artifactId>
+  <version>0.0.0</version>
```
Comment (mziccard) [inline gcloud-java-logging/pom.xml]: What is this version? Where did the parent information go?

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java (hunk 3 above)
Comment (shinfan) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Yes this is actually a known issue. I created a github issue to track this: https://github.com/GoogleCloudPlatform/gcloud-java/issues/1047

**Comment by mziccard** [comment]
We should also add `gcloud-java-logging` to the [parent pom](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/pom.xml#L104) so that tests are triggered.

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java (hunk 1 above)
Comment (shinfan) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Samples added. Done.

File: gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java (hunk 2 above)
Comment (shinfan) [inline gcloud-java-logging/baseline/src/main/java/com/google/cloud/logging/spi/v2/ConfigServiceV2Api.java]: Done.

File: gcloud-java-logging/pom.xml (hunk 4 above)
Comment (shinfan) [inline gcloud-java-logging/pom.xml]: Reverted.

**Comment by shinfan** [comment]
I have updated the parent pom and addressed all the comments. PTAL.

File: gcloud-java-logging/.classpath [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mziccard) [inline gcloud-java-logging/.classpath]: Why this file?

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/v2/testing/LocalLoggingImpl.java [hunk 6]
```diff
@@ -78,22 +76,6 @@ public void listLogEntries(
   }
 
   @Override
-  public void readLogEntries(
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/v2/testing/LocalLoggingImpl.java]: I guess this is no longer needed for the local impl to work, right?

**Comment by mziccard** [comment]
I believe that this PR should also remove gax-related directories (see issue #939).

For what concerns `LocalLoggingImpl` and `LocalLoggingHelper` they should live in the `com.google.cloud.logging.testing` package rather than `com.google.cloud.logging.spi.v2.testing`. However `LocalLoggingImpl` seems a rather simple implementation which is missing several features (support for Config and Metrics sub-services, for instance). I am not sure that we should expose it to users in its current state. @aozarov thoughts?

Are there plans to make `LocalLoggingImpl` a more complete stub or are you waiting for the emulator (see issue #472)? Is there an ETA for the emulator? /cc @jgeewax 

Also just let me remind that there are a bunch of [logging-related issues](https://github.com/GoogleCloudPlatform/gcloud-java/issues?q=is%3Aopen+is%3Aissue+label%3Alogging) that resulted from previous Arie's thorough reviews and that still need to be addressed. If you believe that some of them have already been fixed feel free to close them.

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/v2/testing/LocalLoggingImpl.java (hunk 6 above)
Comment (shinfan) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/spi/v2/testing/LocalLoggingImpl.java]: Right, this method has been removed from LoggingServiceV2Grpc.LoggingServiceV2 so it is no longer needed.

**Comment by shinfan** [comment]
PTAL

I have removed the gax related files and thanks for the reminders. For the <code>LocalLoggingImpl</code> I believe we are waiting for a more scalable testing plan (e.g. emulator but I am sure about the ETA at this point)

Now this PR addressed the following issues:
https://github.com/GoogleCloudPlatform/gcloud-java/issues/943
https://github.com/GoogleCloudPlatform/gcloud-java/issues/771
https://github.com/GoogleCloudPlatform/gcloud-java/issues/939

**Pull request merged**